kingpin/actors/aws/cloudformation.py

```diff
@@ -212,36 +218,63 @@ def _discover_default_params(self, template_body):
         }
         return default_params
 
-    def _get_template_body(self, template):
+    def _get_template_body(self, template: str, s3_region: Optional[str]):
         """Reads in a local template file and returns the contents.
 
         If the template string supplied is a local file resource (has no
         URI prefix), then this method will return the contents of the file.
         Otherwise, returns None.
 
         Args:
-            template: String with a reference to a template location.
 
         Returns:
-            One tuple of:
-              (Contents of template file, None)
-              (None, URL of template)
+          (Contents of template file, None)
+          (Contents of template downloaded from s3, URL of template)
 
         Raises:
             InvalidTemplate
         """
         if template is None:
             return None, None
 
-        remote_types = ('http://', 'https://')
-
-        if template.startswith(remote_types):
-            return None, template
+        if template.startswith('s3://'):
+            match = S3_REGEX.match(template)
+            if match:
+                bucket = match.group('bucket')
+                key = match.group('key')
+            else:
+                raise InvalidTemplate()
+
+            # figure out the region the bucket is in
+            if s3_region is None:
+                log.info(f'Getting region for bucket {bucket}')
+                resp = self.s3_conn.get_bucket_location(Bucket=bucket)
+                s3_region = resp['LocationConstraint']
+                if s3_region is None:
+                    s3_region = 'us-east-1'
+            # AWS support assured me this format would work for all regions for
+            # all buckets they also said
+            # f'https://{bucket}.s3-{region}.amazonaws.com/{key} would also
+            # work but I'm hesitant to use buckets as components in a subdomain
+            url = f'https://s3-{s3_region}.amazonaws.com/{bucket}/{key}'
+
+            s3 = self.get_s3_client(s3_region)
+            try:
+                log.info('Downloading template stored in s3')
+                resp = s3.get_object(Bucket=bucket, Key=key)
+            except Exception as e:
+                raise InvalidTemplate(e)
+            remote_template = resp['Body'].read()
+            return remote_template, url
+        else:
+            # The template is provided inline.
+            try:
+                return json.dumps(self._parse_policy_json(template)), None
+            except exceptions.UnrecoverableActorFailure as e:
+                raise InvalidTemplate(e)
 
-        try:
-            return json.dumps(self._parse_policy_json(template)), None
-        except exceptions.UnrecoverableActorFailure as e:
-            raise InvalidTemplate(e)
+    def get_s3_client(self, region):
```

Comment:
Add a docstring that identifies why we have this function at all - why you would use this instead of the built-in s3 client object for the actor..